src/Common/UpgradeNotification/UpgradeNotificationHelper.cs

```diff
@@ -0,0 +1,259 @@
+                {
+                    return;
+                }
+
+                //has done check this session, skip
+                if (hasNotified)
+                {
+                    return;
+                }
+
+                //register verion check and upgrade notification in frequency service
+                if (frequencyService == null) {
+                    return;
+                }
+                frequencyService.Add(FrequencyKeyForUpgradeCheck, FrequencyTimeSpanForUpgradeCheck);
+                frequencyService.Add(FrequencyKeyForUpgradeNotification, FrequencyTimeSpanForUpgradeNotification);
+
+                string checkModuleName = "Az";
+                string checkModuleCurrentVersion = _qosEvent.AzVersion;
+                string upgradeModuleNames = "Az";
+                if ("0.0.0" == _qosEvent.AzVersion)
+                {
+                    checkModuleName = _qosEvent.ModuleName;
+                    checkModuleCurrentVersion = _qosEvent.ModuleVersion;
+                    upgradeModuleNames = "Az.*";
+                }
+
+                //refresh az module versions if necessary
+                frequencyService.Check(FrequencyKeyForUpgradeCheck, () => true, () =>
+                {
+                    Thread loadHigherVersionsThread = new Thread(new ThreadStart(() =>
+                    {
+                        _qosEvent.HigherVersionsChecked = true;
+                        try
+                        {
+                            //no lock for this method, may skip some notifications, it's expected.
+                            RefreshVersionInfo(upgradeModuleNames);
+                        }
+                        catch (Exception)
+                        {
+                            //do nothing
+                        }
+                    }));
+                    loadHigherVersionsThread.Start();
+                });
+
+                bool shouldPrintWarningMsg = HasHigherVersion(checkModuleName, checkModuleCurrentVersion);
+
+                //prompt warning message for upgrade if necessary
+                frequencyService.Check(FrequencyKeyForUpgradeNotification, () => shouldPrintWarningMsg, () =>
+                {
+                    _qosEvent.UpgradeNotificationPrompted = true;
+                    hasNotified = true;
+
+                    string latestModuleVersion = GetModuleLatestVersion(checkModuleName);
+                    string updateModuleCmdletName = GetCmdletForUpdateModule();
+                    string warningMsg = $"You're using {checkModuleName} version {checkModuleCurrentVersion}. The latest version of {checkModuleName} is {latestModuleVersion}. Upgrade your Az modules using the following commands:\n";
+                    warningMsg += $"  {updateModuleCmdletName} {upgradeModuleNames} -WhatIf \t -- Simulate updating your Az modules.\n";
+                    warningMsg += $"  {updateModuleCmdletName} {upgradeModuleNames} \t\t -- Update your Az modules.\n";
```

Comment:
```suggestion
                    warningMsg += $"  {updateModuleCmdletName} {upgradeModuleNames} \t\t -- Update your Az modules.{Environment.NewLine}";
```
- Use Environment.NewLine to be compatible in different platforms.
- Use 4 spaces to replace \t to avoid different width in different platforms.
- Could you provide a picture demonstrate how the warning messages are displayed in console?